src/protocol.c

```diff
@@ -136,8 +138,16 @@ thread_run_command(void *args) {
                 perror("setenv");
                 exit(1);
             }
-            if (execvp(server->argv[0], server->argv) < 0) {
-                perror("execvp");
+            int e = -1;
+            if ( 0 == access(server->argv[0], R_OK | X_OK) ) {
+                e = execvp(server->argv[0], server->argv);
+            } else {
+                char *argp[] = {"sh", "-c", NULL, NULL};
+                argp[2] = server->argv[0];
+                e = execv("/bin/sh", argp);
```

Comment:
you cannot use the command line without changing it or you need to do some fancy analysis of the quoting imagine "strange 'command' --with='-unexpected argument'"

This code is straight out the system() libc code.